services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusHistoryService.kt

```diff
@@ -0,0 +1,168 @@
+package fi.hel.haitaton.hanke.hakemus
+
+import fi.hel.haitaton.hanke.allu.AlluStatusRepository
+import fi.hel.haitaton.hanke.allu.ApplicationHistory
+import fi.hel.haitaton.hanke.allu.ApplicationStatus
+import fi.hel.haitaton.hanke.allu.ApplicationStatusEvent
+import fi.hel.haitaton.hanke.email.InformationRequestEmail
+import fi.hel.haitaton.hanke.email.JohtoselvitysCompleteEmail
+import fi.hel.haitaton.hanke.email.KaivuilmoitusDecisionEmail
+import fi.hel.haitaton.hanke.paatos.PaatosService
+import fi.hel.haitaton.hanke.permissions.HankeKayttajaService
+import fi.hel.haitaton.hanke.permissions.PermissionCode
+import fi.hel.haitaton.hanke.taydennys.TaydennysService
+import java.time.OffsetDateTime
+import mu.KotlinLogging
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+private val logger = KotlinLogging.logger {}
+
+@Service
+class HakemusHistoryService(
+    private val hakemusRepository: HakemusRepository,
+    private val alluStatusRepository: AlluStatusRepository,
+    private val taydennysService: TaydennysService,
+    private val paatosService: PaatosService,
+    private val hankeKayttajaService: HankeKayttajaService,
+    private val applicationEventPublisher: ApplicationEventPublisher,
+) {
+
+    @Transactional(readOnly = true) fun getAllAlluIds() = hakemusRepository.getAllAlluIds()
+
+    @Transactional(readOnly = true)
+    fun getLastUpdateTime() = alluStatusRepository.getLastUpdateTime()
+
+    @Transactional
+    fun handleHakemusUpdates(
+        applicationHistories: List<ApplicationHistory>,
+        updateTime: OffsetDateTime
+    ) {
+        applicationHistories.forEach { handleHakemusUpdate(it) }
+        val status = alluStatusRepository.getReferenceById(1)
+        status.historyLastUpdated = updateTime
+        alluStatusRepository.save(status)
```

Comment:
Is explicite save necessary for the entity inside a transaction?

Reply:
I guess it isn't. Removed.